www/class/centreonCustomView.class.php

```diff
@@ -640,9 +640,60 @@ public function loadCustomView($params)
         if (!$dbResult) {
             throw new \Exception("An error occured");
         }
+        
+        //if the view is being added for the first time, we make sure that the widget parameters are going to be set
+        if (!$update) {
+            $this->addPublicViewWidgetParams($params['viewLoad'], $this->userId);
+        }
+        
         return $params['viewLoad'];
     }
 
+    /**
+    * @param $viewId
+    * @param $userId
+    * @throws Exception
+    */
+    public function addPublicViewWidgetParams($viewId, $userId)
+    {
+        //get all widget parameters from the view that is being added
+        if (isset($userId) && $userId) {
+            $stmt = $this->db->prepare(
+                'SELECT * FROM widget_views wv ' .
+                'LEFT JOIN widget_preferences wp ON wp.widget_view_id = wv.widget_view_id ' .
+                'LEFT JOIN custom_view_user_relation cvur ON cvur.custom_view_id=wv.custom_view_id ' .
+                'WHERE cvur.custom_view_id = :viewId and cvur.locked = 0'
```

Comment:
'WHERE cvur.custom_view_id = :view_id AND cvur.is_owner = 1 AND cvur.user_id=wp.user_id'

the query wasn't specific enough (if multiple users had the view shared via the shared button, duplicated each parameters for each user having the view shared to them). And as I expected it to be.  If we share the view to an unlocked user we create a bug.

Reply:
done ;)
I corrected some other psr2 style and a missing "$" too